service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    // get all user tokens/uuids
+    @Override
+    public List<DataAccessToken> getAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        return allDataAccessTokens;
+    }
+    
+    // get data access token for a given username
+    // get all data access tokens available for user in table and return newest one (uuid)
+    @Override
+    public DataAccessToken getDataAccessToken(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken newestDataAccessToken = getNewestDataAccessToken(allDataAccessTokens);
+        return newestDataAccessToken;
+    }
+
+    @Override
+    public DataAccessToken getDataAccessTokenInfo(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken;
+    }
+
+    @Override
+    public void revokeAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        for (DataAccessToken dataAccessToken : allDataAccessTokens) {
+            dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+        } 
+    }
+
+    @Override
+    public void revokeDataAccessToken(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+    }
+    
+    @Override
+    public String getUsername(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getUsername();
+    }
+    
+    @Override
+    public Date getExpiration(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getExpiration();
+    }
+   
+    @Override
+    public Boolean isValid(String dataAccessToken) {
+        DataAccessToken storedDataAccessToken = null;
+        try {
+            storedDataAccessToken = dataAccessTokenRepository.getDataAccessToken(dataAccessToken);
+        } catch (Exception e) {
+            log.error("Error retrieving data access token, " + dataAccessToken + " from token store");
+            return Boolean.FALSE;
+        }
+        Calendar calendar = Calendar.getInstance();
+        Date currentDate = calendar.getTime();
+        if (storedDataAccessToken.getExpiration().before(currentDate)) {
+            return Boolean.FALSE;
+        } 
+        return Boolean.TRUE;
+    }
+
+    private int getNumberOfTokensForUsername(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
```

Comment:
This can be made into a custom query which just returns the results of select count(*) from data_access_token where username = `$?` .. that way less data is passed around and ultimately ignored